packages/db/tests/lib/test-QuerySearchClient.js

```diff
@@ -0,0 +1,256 @@
+  await destroyLocalTestDb({
+    ...t.context,
+  });
+});
+
+const createFileRecords = async ({
+  granuleCumulusId,
+  filePgModel,
+  knex,
+  bucket,
+}, numOfFileRecords) => {
+  const insertRecords = [...new Array(numOfFileRecords).keys()]
+    .map((index) => ({
+      bucket,
+      key: `${index}_${cryptoRandomString({ length: 5 })}`,
+      granule_cumulus_id: granuleCumulusId,
+    }));
+  const records = await filePgModel.insert(
+    knex,
+    insertRecords,
+    '*'
+  );
+  return records;
+};
+
+test('QuerySearchClient.shift() returns next record from current set of results correctly', async (t) => {
+  const { knex, bucket, testGranule } = t.context;
+
+  const records = orderBy(await createFileRecords(t.context, 2), ['cumulus_id']);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const querySearchClient = new QuerySearchClient(
+    query,
+    5
+  );
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[0],
+      granule_cumulus_id: Number.parseInt(records[0].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[1],
+      granule_cumulus_id: Number.parseInt(records[1].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+});
+
+test('QuerySearchClient.shift() returns next record if next record must be fetched', async (t) => {
+  const { knex, bucket, testGranule } = t.context;
+
+  const records = orderBy(await createFileRecords(t.context, 2), ['cumulus_id']);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const querySearchClient = new QuerySearchClient(
+    query,
+    1
+  );
+  const queryOffsetSpy = sinon.spy(query, 'offset');
+  const queryLimitSpy = sinon.spy(query, 'limit');
+
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[0],
+      granule_cumulus_id: Number.parseInt(records[0].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.like(
+    await querySearchClient.shift(),
+    {
+      ...records[1],
+      granule_cumulus_id: Number.parseInt(records[1].granule_cumulus_id, 10),
+      granule_id: testGranule.granule_id,
+    }
+  );
+  t.is(queryOffsetSpy.callCount, 2);
+  t.is(queryLimitSpy.callCount, 2);
+});
+
+test('QuerySearchClient.peek() correctly returns true if next record exists in fetched results', async (t) => {
+  const { knex, bucket } = t.context;
+
+  await createFileRecords(t.context, 1);
+
+  const query = getFilesAndGranuleInfoQuery({
+    knex,
+    searchParams: { bucket },
+    sortColumns: ['cumulus_id'],
+    granuleColumns: ['granule_id'],
+  });
+  const fileSearchClient = new QuerySearchClient(
+    query,
+    1
+  );
+  t.truthy(
```

Comment:
Nit but maybe avoid `truthy` and update the test statement because `peek()` doesn't technically return `true`